runtime/lua/vim/fs.lua

```diff
@@ -348,7 +348,11 @@ function M.normalize(path, opts)
     path = path:gsub('%$([%w_]+)', vim.uv.os_getenv)
   end
 
-  return (path:gsub('\\', '/'):gsub('/+', '/'):gsub('(.)/$', '%1'))
+  path = path:gsub('\\', '/'):gsub('/+', '/')
+  if iswin and path:match('^%w:/$') then
+    return path
+  end
+  return (path:gsub('(.)/$', '%1'))
```

Comment:
Why don't we always preserve the last slash instead of only on Windows? Based on previous discussion https://github.com/neovim/neovim/pull/23937#discussion_r1219692472 it doesn't sound like there was strong opposition. And that would avoid this special-case. @gpanders @lewis6991 @clason

Reply:
I guess this was part of 0.9 so back-compat is a concern. But normalize() is much more useful if its result is very predictable.